specifyweb/specify/management/commands/check_locked_tables.py

```diff
@@ -0,0 +1,42 @@
+from django.core.management.base import BaseCommand
+from django.db import connection
+from time import sleep
+
+def check_locked_tables(table_names=None):
```

Comment:
Ya, what I really want to figure out is how this issue might be causing Django to crash.

Reply:
I might be open to adjusting database parameters in a case like this.  This seems to be a rare case, so clients running their own databases might still run into this issue, although I don't think there have been any reports of it.  Right now, it's more import to make sure the specify cloud instances don't run into this problem.  Recreating the problem on the test server might be a good idea, to see if we can crash a Django instance.

Reply:
I remember you saying there were 20 connections. I suspect they were all updates, being queued at the database. We also have just a dual core on the canada server. I was going to test this out last night, but I ran out of time. I was going to upload a workbench dataset on the sp7demofish-canada, while making tree merge requests. If it is just a django issue (20 connections overloading), then the size of database wouldn't matter. If the django instance crashes, we'll know that it is because of connection overload.